### PR TITLE
Optional logging to remote syslog

### PR DESCRIPTION
**Pull request by ChozeTM**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,26 @@ volumes:
   - /your_local_path:/var/log/
 ```
 
+You can also forward log output to some remote syslog server, like Graylog. It always uses UDP protocol and port 514 as default value,
+port number can be changed to another one.
+
+```
+environment:
+  ...
+  - RSYSLOG_REMOTE_IP=my.remote-syslog-server.com
```
Comment (wader) [inline README.md:94]: Maybe `RSYSLOG_REMOTE_HOST` is clearer?

File: README.md [hunk 2]
```diff
@@ -85,6 +85,26 @@ volumes:
...
+  - RSYSLOG_REMOTE_PORT=5140
+```
+
+### Timezone
+Wrong timestamps in log can be fixed by setting proper timezone.
+This parameter is handled by source Debian image.
```
Comment (wader) [inline README.md:100]: You mean source as in debian base image?

File: README.md [hunk 3]
```diff
@@ -85,6 +85,26 @@ volumes:
...
+
+```
+environment:
+  ...
+  - TZ=Europe/Prague
```
Comment (wader) [inline README.md:105]: Just be sure, timezones are available now in the image? no need to install some additional package?

File: run [hunk 4]
```diff
@@ -89,6 +89,13 @@ EOF
   if [ "${RSYSLOG_LOG_TO_FILE}" == 'yes' ] ; then
     echo 'mail.* -/var/log/mail.log' >> /etc/rsyslog.conf
   fi
+
+  if [ ! ${RSYSLOG_REMOTE_PORT} ] ; then
+    RSYSLOG_REMOTE_PORT=514
+  fi
+  if [ ${RSYSLOG_REMOTE_IP} ] ; then
```
Comment (wader) [inline run:96]: Above we use `-z` to check for empty value, i guess this works also? but maybe nice to be consistent

File: run [hunk 5]
```diff
@@ -89,6 +89,13 @@ EOF
...
+    echo "*.* action(type=\"omfwd\" target=\"${RSYSLOG_REMOTE_IP}\" port=\"${RSYSLOG_REMOTE_PORT}\" template=\"RSYSLOG_SyslogProtocol23Format\")" >> /etc/rsyslog.conf
```
Comment (wader) [inline run:97]: I don't know much about syslog protocols, do ppl use other protocols like `RSYSLOG_ForwardFormat`?

**Review by wader — COMMENTED** [review]
Nice, looks good, just some small nitpicks and comments

File: README.md (hunk 3 above)
Comment (ChozeTM) [reply README.md:105]: Yes, i've just added TZ to docker-compose.yml and timezone has changed. I found this info in some forum.

File: README.md (hunk 2 above)
Comment (ChozeTM) [reply README.md:100]: Yes, Debian base image.

File: run (hunk 4 above)
Comment (ChozeTM) [reply run:96]: You're right.

File: README.md (hunk 1 above)
Comment (ChozeTM) [reply README.md:94]: agree

File: run (hunk 5 above)
Comment (ChozeTM) [reply run:97]: I have to check it and test first.  RSYSLOG_SyslogProtocol23Format  gave me nice output in Graylog, it has split message to separate columns out of the box, no need to parse it.
image

Comment (wader) [reply run:97]: Ok, think i'm fine either hardcode or have an option with default RSYSLOG_SyslogProtocol23Format

**Comment by wader** [comment]
Let me know when you feel ready to merge

**Comment by ChozeTM** [comment]
I did changes as you advised, plus added extra parameter for remote syslog format. I am ready to go.

**Comment by wader** [comment]
Thanks, did you want RSYSLOG_ForwardFormat to be default or RSYSLOG_SyslogProtocol23Format?

**Comment by ChozeTM** [comment]
I made RSYSLOG_ForwardFormat as default, seems to be prefered according to manpages.

**Comment by wader** [comment]
Merging, thanks! will make a new tag also

**Pull request merged**